### PR TITLE
Remove wrong type-hints and stale docstrings from distributions

### PR DESCRIPTION
**Pull request by ricardoV94**
Related to #4859

This was originally proposed in #6272, see https://github.com/pymc-devs/pymc/pull/6272#pullrequestreview-1172309680

## Major / Breaking Changes
- ...

## Bugfixes / New features
- ...

## Docs / Maintenance
- Remove stale `logp` and `logcdf` docstrings

### CODE REVIEW

**Comment by canyon289** [comment]
Thank you Ricardo! I'll get to this in next 96 hours

**Comment by ricardoV94** [comment]
> Not blocking the PR, but trying to think forawrd. How do we expect users to know what parameters they need to pass for logp/logcdf to work? several distributions also have alternative parametrizations

Users should never call these methods themselves. The user-facing API is `pm.logp(rv, value)`

**Comment by ricardoV94** [comment]
I would bet a newbie developer would be more confused than enlightened by these broken docstrings...

File: pymc/distributions/continuous.py [hunk 1]
```diff
@@ -1051,31 +925,7 @@ def get_mu_lam_phi(
...
-        Calculate log-probability of Wald distribution at specified value.
-
-        Parameters
-        ----------
-        value : tensor_like of float
-            Value(s) for which log-probability is calculated. If the log probabilities for multiple
-            values are desired the values must be provided in a numpy array or Aesara tensor.
-        mu : tensor_like of float
-            Mean of the distribution (mu > 0).
-        lam : tensor_like of float
```
Comment (canyon289) [inline pymc/distributions/continuous.py:1070]: This is a really good example. I wouldnt know what lam or alpha are by memory and from the logp calculation below theres not enough math to tell me. Having even just a basic description of the parameters would be helpful mathematically.

The other is the what is allowed as inputs. If I was to write code that called this logp I would know what to pass, and auto type checking would be useful.

In the scope of things I care way more about 1 than I do 2, though both would be nice

Comment (ricardoV94) [reply pymc/distributions/continuous.py:1070]: This is a really good example of why I don't like it. The return type is completely wrong, should be TensorVariable (Type) not RandomVariable (Op). Also we only test and promise to support TensorVariable inputs, not float or np.ndarray (even though many times they will work)

Comment (ricardoV94) [reply pymc/distributions/continuous.py:1070]: The parameter information is contained in the main distribution docstring already.

Comment (ricardoV94) [reply pymc/distributions/continuous.py:1070]: https://github.com/pymc-devs/pymc/blob/d34b557b5e0a0806842e2d4ec5aa5a842dbf8dfe/pymc/distributions/continuous.py#L861-L868

**Comment by maresb** [comment]
Ok, presumably with the pin in #6282, mypy is now passing. Is this ready to merge? (I have not reviewed.)

**Review by OriolAbril — APPROVED** [review]

**Pull request merged**